ewcasino.py

```diff
@@ -1314,6 +1319,234 @@
+			else:
+				countdown_resp = "{}{}{}".format(text_mod, countdown, text_mod)
+				
+			countdown -= 1
+			await ewutils.send_message(cmd.client, cmd.message.channel, countdown_resp)
+
+			try:
+				msg = await cmd.client.wait_for_message(timeout=1, author=author)
+
+				if msg != None:
+					
+					challenger.rr_challenger = ""
+					challengee.rr_challenger = ""
+
+					challenger.rr_restriction = 0
+					challengee.rr_restriction = 0
+
+					challengee.persist()
+					
+					die_resp = challenger.die(cause=ewcfg.cause_praying)
+					challenger.persist()
+					await ewrolemgr.updateRoles(client=cmd.client, member=author)
+					await die_resp.post()
+
+					response = "ENDLESS WAR completely and utterly obliterates {} with a bone-hurting beam. The duel is over before it even began.".format(author.display_name).replace("@","\{at\}")
+					
+					return await ewutils.send_message(cmd.client, cmd.message.channel, response)
+
+
+			except:
+				pass
+
+			try:
+				msg = await cmd.client.wait_for_message(timeout=1, author=member)
+
+				if msg != None:
+					challenger.rr_challenger = ""
+					challengee.rr_challenger = ""
+
+					challenger.rr_restriction = 0
+					challengee.rr_restriction = 0
+
+					challenger.persist()
+
+					die_resp = challengee.die(cause=ewcfg.cause_praying)
+					challengee.persist()
+					await ewrolemgr.updateRoles(client=cmd.client, member=member)
+					await die_resp.post()
+
+					response = "ENDLESS WAR completely and utterly obliterates {} with a bone-hurting beam. The duel is over before it even began.".format(member.display_name).replace("@", "\{at\}")
+
+					return await ewutils.send_message(cmd.client, cmd.message.channel, response)
+
+			except:
+				pass
+			
+		response = "{}***DRAW!***{}".format(ewcfg.emote_slimegun, ewcfg.emote_slimegun)
+		await ewutils.send_message(cmd.client, cmd.message.channel, response)
+		
+		# start the duel
+		challenger.time_expirpvp = ewutils.calculatePvpTimer(challenger.time_expirpvp, (int(time.time()) + ewcfg.time_pvp_duel))
+		challengee.time_expirpvp = ewutils.calculatePvpTimer(challengee.time_expirpvp, (int(time.time()) + ewcfg.time_pvp_duel))
+
+		challenger.persist()
+		challengee.persist()
+
+		await ewrolemgr.updateRoles(client=cmd.client, member=author)
+		await ewrolemgr.updateRoles(client=cmd.client, member=member)
+		
+		duel_timer = ewcfg.time_pvp_duel
+		
+		challenger_slimes = challenger.slimes
+		challengee_slimes = challengee.slimes
+		
+		while challenger_slimes > 0 and challengee_slimes > 0 and duel_timer > 0:
+			# count down from 2 minutes, the max possible time a duel should last
+			duel_timer -= 1
+
+			# re-grab slime counts from both duelers
+			challenger = EwUser(member=author)
+			challengee = EwUser(member=member)
+			challenger_slimes = challenger.slimes
+			challengee_slimes = challengee.slimes
+
+			await asyncio.sleep(1)
+			
+		if challenger.slimes <= 0:
```

Comment:
reload user data after await call yadda yadda, this probably happens more times in this method